commit project

garini491 opened the pull request.

garini491 committed:
commit project

pomegranate282922 commented on src/main/java/pages/ScooterStartPage.java:
```diff
@@ -0,0 +1,70 @@
+    }
+    // Метод нажатия на кнопку куки
+    public void coockyButtonClick() {
+        driver.findElement(coockyButton).click();
```
cookie

pomegranate282922 commented on src/main/java/pages/ScooterOrderPage.java:
```diff
@@ -0,0 +1,179 @@
+
+    // Метод выюора самоката
+    public void setScooter (String color) {
+        if(color == "чёрный жемчуг") {
```
Строки сравниваем через equals

pomegranate282922 commented on src/test/java/pages/ScooterOrderPageTest.java:
```diff
@@ -0,0 +1,137 @@
+    }
+
+    @After
+    public void tierDown() {
```
tear

pomegranate282922 commented on src/test/java/pages/ScooterOrderPageTest.java:
```diff
@@ -0,0 +1,137 @@
+        createOrder.regScooter(day,period,color,comment);
+        createOrder.yesClick();
+        String expected = "Заказ оформлен";
+        assertEquals(expected,createOrder.orderCreated());
```
Здесь нужно перепроверить реализацию. В хроме тест должен падать, а в ff быть зелёным.

pomegranate282922 commented on src/test/java/pages/ScooterStartPageTest.java:
```diff
@@ -0,0 +1,58 @@
+        startPage.scrollToQuestions();
+        for (int i = 0; i < 8; i++) {
+            driver.findElement(By.xpath(".//div[@id = 'accordion__heading-" + i + "']")).click();
+            WebElement element = driver.findElement(By.xpath(".//div[@id = 'accordion__panel-" + i +"']/p"));
```
Page object model предполагает, что на уровне тестов у нас нет локаторов. Их нужно перенести на уровень page object, а здесь вызывать только методы из po

pomegranate282922 reviewed (commented):
Хорошая работа! Нужно немного доработать page object model, поправить мелкие огрехи по коду.

garini491 committed:
поправил орфографические ошибки.
Перенес локаторы в po классы
переделал реализацию проверки создания заказа
убрал полноценную отмену заказа, т.к. не смог разобраться с кнопкой "Хорошо". у кнопки по факту локатор тот же что и у остальных на ее месте, но по тексту ее не находит.
Так же заметил что иногда не приходит номер заказа.

pomegranate282922 commented on src/test/java/pages/ScooterStartPageTest.java:
```diff
@@ -0,0 +1,56 @@
+        startPage.openStartPage();
+        startPage.cookieButtonClick();
+        startPage.scrollToQuestions();
+        for (int i = 0; i < 8; i++) {
```
Здесь в коде получается нарушение принципа один тест - одна проверка. Нужно или подумать над параметризацией, или сделать явные отдельные тестовые методы на каждый пункт.

pomegranate282922 commented on src/test/java/pages/ScooterOrderPageTest.java:
```diff
@@ -0,0 +1,139 @@
+        createOrder.regData(name,lname,adres,station,phone);
+        createOrder.regScooter(day,period,color,comment);
+        createOrder.yesClick();
+        assertEquals(createOrder.getExpected(),createOrder.orderCreated());
```
Здесь нужно проверить, что появилось окошко с надписью Заказ оформлен.

pomegranate282922 reviewed (commented):
Я прошу прощения - в первом ревью не заметила важной детали в faq тестах - в них получился один тест на все проверки, что нарушает принцип один тест - одна проверка.
Также осталась не совсем корректная проверка про создании заказа. Оставила комментарий, что нужно сделать.

garini491 committed:
переделал тесты вопросов/ответов на параметризованные:
убрал массив с ответами, перенес их в параметры.
переделал проверку на отображение окна с сообщением, что заказ оформлен успешно

pomegranate282922 reviewed (approved):
Зачёт!
